public/app/features/alerting/unified/hooks/mergeRequestStates.ts

```diff
@@ -0,0 +1,24 @@
+interface RequestState {
+  error?: unknown;
+
+  isUninitialized: boolean;
+  isSuccess: boolean;
+  isError: boolean;
+  isLoading: boolean;
+}
+
+export interface AggregateRequestState extends RequestState {
+  requests: RequestState[];
+}
+
+// @TODO what to do with the other props that we get from RTKQ's state such as originalArgs, etc?
+export function mergeRequestStates(...requests: RequestState[]): AggregateRequestState {
```

Comment:
This is no longer used after the refactors to manage the state manually

Reply:
I copied this one from the other PR on managing Alertmanager entities where we can use this, I'll remove it from this PR but it will be back in some form later :D